lib/pf/web/device_registration.pm

```diff
@@ -18,65 +18,95 @@ use warnings;
 use HTML::Entities;
 use Readonly;
 
-use pf::config;
-use pf::file_paths qw($allowed_device_oui_file $allowed_device_types_file);
+use pf::constants;
+use pf::config qw(%Config);
 use pf::enforcement qw(reevaluate_access);
 use pf::node qw(node_register is_max_reg_nodes_reached);
 use pf::util;
+use pf::error qw(is_success);
+use pf::log;
 use pf::web;
 use pf::web::custom;    # called last to allow redefinitions
 
 use pf::authentication;
 use pf::Authentication::constants;
 use List::MoreUtils qw(any);
 
-Readonly our @DEVICE_OUI => _load_file_into_array($allowed_device_oui_file);
-Readonly our @DEVICE_TYPES => _load_file_into_array($allowed_device_types_file);
-
 =head1 SUBROUTINES
```

Comment:
A :nail_care: comment:
We're trying to move away from the lib/pf/web/ directory and remove things that are in it.
I think your code for the device validation would be better placed in the DeviceRegistration controller